pom.xml

```diff
@@ -1,164 +1,1087 @@
+    </dependency>
+    <dependency>
+      <groupId>com.mattbertolini</groupId>
+      <artifactId>liquibase-slf4j</artifactId>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.jmock</groupId>
+      <artifactId>jmock</artifactId>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.jmock</groupId>
+      <artifactId>jmock-junit4</artifactId>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.hsqldb</groupId>
+      <artifactId>hsqldb</artifactId>
+      <scope>test</scope>
+    </dependency>
+  </dependencies>
+  <repositories>
+    <repository>
+      <id>jboss</id>
+      <name>JBoss</name>
+      <url>https://repository.jboss.org/nexus/content/groups/public/</url>
+    </repository>
+    <repository>
+      <id>oauth</id>
+      <name>Oauth</name>
+      <url>https://oauth.googlecode.com/svn/code/maven/</url>
+    </repository>
+    <repository>
+      <id>awood</id>
+      <name>awood.fedorapeople.org</name>
+      <url>https://awood.fedorapeople.org/ivy/candlepin/</url>
+    </repository>
+    <repository>
+      <id>barnabycourt</id>
+      <name>barnabycourt.fedorapeople.org</name>
+      <url>https://barnabycourt.fedorapeople.org/repo/candlepin/</url>
+    </repository>
+  </repositories>
+  <build>
+    <resources>
+      <resource>
+        <filtering>true</filtering>
+        <directory>src/main/resources</directory>
+      </resource>
+      <resource>
+        <filtering>true</filtering>
+        <directory>${project.build.directory}/generated-resources</directory>
+      </resource>
+    </resources>
+    <testResources>
+      <testResource>
+        <filtering>false</filtering>
+        <directory>src/test/resources</directory>
+      </testResource>
+    </testResources>
+    <plugins>
+      <plugin>
+        <groupId>org.openapitools</groupId>
+        <artifactId>openapi-generator-maven-plugin</artifactId>
+        <version>5.1.0</version>
+        <executions>
+          <execution>
+            <goals>
+              <goal>generate</goal>
+            </goals>
+            <configuration>
+              <inputSpec>${project.basedir}/api/candlepin-api-spec.yaml</inputSpec>
+              <generatorName>jaxrs-spec</generatorName>
+              <configurationFile>${project.basedir}/api/candlepin-api-config.json</configurationFile>
+              <configOptions>
+                <interfaceOnly>true</interfaceOnly>
+                <generatePom>false</generatePom>
+                <dateLibrary>java8</dateLibrary>
+                <sourceFolder>src/gen/java/main</sourceFolder>
+              </configOptions>
+              <templateDirectory>${project.basedir}/buildSrc/src/main/resources/templates</templateDirectory>
+            </configuration>
+          </execution>
+        </executions>
+      </plugin>
+      <plugin>
+        <groupId>org.bsc.maven</groupId>
+        <artifactId>maven-processor-plugin</artifactId>
```

Comment:
A question on these 2 plugins: maven-processor-plugin and build-helper-maven-plugin. They didn't seem to be needed beforehand; why are they added now? How where the jpamodel generated-sources added to the war during a maven build before?

Reply:
Since we are using open-api & jpamodel generators, both generate source code to ```/target/generated-sources``` directory, which creates conflicts. We need to explicitly specify where to keep the generated code (by specifying exact output directory). ```maven-processor-plugin``` (& ```openapi-generator-maven-plugin```) plugin makes job easier by specifying output directory.

```build-helper-maven-plugin``` make sure that multiple source directories (auto-generated & main code) are available during compilation.
The old candlepin project structure, every sub-project (server, common, api) have specific POM file, which does not create any conflict while auto-generating code since each project have their own ```target/generated-sources``` directory.